PhysicsTools/PatAlgos/python/slimming/customizeMiniAOD_MuEGFixMoriond2017.py

```diff
@@ -0,0 +1,138 @@
+    process.load("RecoBTag.SoftLepton.softLepton_cff")
+    process.load("RecoBTag.Combined.combinedMVA_cff")
+    process.load("RecoBTag.CTagging.cTagging_cff")
+    process.load("RecoVertex.AdaptiveVertexFinder.inclusiveVertexing_cff")
+
+def customizeAll(process, verbose=False):
+    
+    process = customizeGSFixForPAT(process)
+
+    loadJetMETBTag(process)
+    backupJetSequences = backupJetsFirstStep(process)
+
+    addBadMuonFilters(process)    
+    badMuons = cms.VInputTag( cms.InputTag("badGlobalMuonTagger","bad"), cms.InputTag("cloneGlobalMuonTagger","bad") )
+    # clean the muons and PF candidates, and make *everything* point to the new candidates
+    cleanPFCandidates(process, badMuons, verbose=verbose) 
+
+    addDiscardedPFCandidates(process, cms.InputTag("pfCandidatesBadMuonsCleaned","discarded"), verbose=verbose)
+
+    # now make the backup sequences point to the right place
+    backupJetsSecondStep(process, backupJetSequences, badMuons, verbose=verbose)
+
+    
+    process.patMuons.embedCaloMETMuonCorrs = False # FIXME
+    ##extra METs and MET corrections ===============================================================
+    from PhysicsTools.PatAlgos.slimming.extraSlimmedMETs_MuEGFixMoriond2017 import addExtraMETCollections,addExtraPuppiMETCorrections
+    
+    addExtraMETCollections(process,
+                           unCleanPFCandidateCollection="particleFlow",
+                           cleanElectronCollection="slimmedElectrons",
+                           cleanPhotonCollection="slimmedPhotons",
+                           unCleanElectronCollection="slimmedElectronsBeforeGSFix",
+                           unCleanPhotonCollection="slimmedPhotonsBeforeGSFix")
+    addExtraPuppiMETCorrections(process,
+                                cleanPFCandidateCollection="particleFlow",
+                                unCleanPFCandidateCollection="pfCandidatesBadMuonsCleaned",
+                                cleanElectronCollection="slimmedElectrons",
+                                cleanPhotonCollection="slimmedPhotons",
+                                unCleanElectronCollection="slimmedElectronsBeforeGSFix",
+                                unCleanPhotonCollection="slimmedPhotonsBeforeGSFix")
+
+    addKeepStatement(process,
+                     "keep *_slimmedMETs_*_*",
+                     ["keep *_slimmedMETsUncorrected_*_*",
+                      "keep *_slimmedMETsEGClean_*_*",
+                      "keep *_slimmedMETsMuEGClean_*_*"],
+                     verbose=verbose)
+    addKeepStatement(process,
+                     "keep *_slimmedMETsPuppi_*_*",
+                     ["keep *_puppiMETEGCor_*_*",
+                      "keep *_puppiMETMuCor_*_*"],
+                     verbose=verbose)
+
+
+
+    #redo the miniAOD data customization for new JEC modules created during the backup process
+    from PhysicsTools.PatAlgos.slimming.miniAOD_tools import miniAOD_customizeData
+    miniAOD_customizeData(process)
```

Comment:
@slava77 I did not noticed any configuration issue by calling it twice, but in case there is something we missed it can be replaced. The second miniAOD_customizeData is needed to ensure that residuals JECs are properly taken into account in the new METs.